opentelemetry-instrumentation-tower/src/lib.rs

```diff
@@ -249,41 +269,61 @@ impl<ReqExt, ResExt> HTTPMetricsLayerBuilder<ReqExt, ResExt> {
     pub fn with_response_extractor_fn<F, B>(
         self,
         f: F,
-    ) -> HTTPMetricsLayerBuilder<ReqExt, FnResponseExtractor<F>>
+    ) -> HTTPLayerBuilder<ReqExt, FnResponseExtractor<F>>
     where
         F: Fn(&http::Response<B>) -> Vec<KeyValue> + Clone + Send + Sync + 'static,
     {
         self.with_response_extractor(FnResponseExtractor::new(f))
     }
 
-    pub fn build(self) -> Result<HTTPMetricsLayer<ReqExt, ResExt>> {
+    pub fn build(self) -> Result<HTTPLayer<ReqExt, ResExt>> {
         let req_dur_bounds = self
             .req_dur_bounds
             .unwrap_or_else(|| LIBRARY_DEFAULT_HTTP_SERVER_DURATION_BOUNDARIES.to_vec());
-        match self.meter {
-            Some(meter) => Ok(HTTPMetricsLayer {
-                state: Arc::from(Self::make_state(meter, req_dur_bounds)),
-                request_extractor: self.request_extractor,
-                response_extractor: self.response_extractor,
-            }),
-            None => Err(Error {
-                inner: ErrorKind::Config(String::from("no meter provided")),
-            }),
-        }
-    }
 
-    pub fn with_meter(mut self, meter: Meter) -> Self {
-        self.meter = Some(meter);
-        self
+        Ok(HTTPLayer {
+            state: Arc::from(Self::make_state(
+                self.meter_provider,
+                self.tracer_provider,
+                req_dur_bounds,
+            )),
+            request_extractor: self.request_extractor,
+            response_extractor: self.response_extractor,
+        })
     }
 
     pub fn with_request_duration_bounds(mut self, bounds: Vec<f64>) -> Self {
         self.req_dur_bounds = Some(bounds);
         self
     }
 
-    fn make_state(meter: Meter, req_dur_bounds: Vec<f64>) -> HTTPMetricsLayerState {
-        HTTPMetricsLayerState {
+    /// Set a meter provider to use for creating a meter.
+    /// If none is specified, the global provider is used.
```

Comment:
lets just use the global provider always. If there is a concrete reason why global cannot work, then we can follow up and accept a provider. (If we ever need to do that, it just need to accept a tracer or meter. Not the provider itself.But thats for another time discussion)